packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-input.tsx

```diff
@@ -94,8 +99,21 @@ const TakeProfitAndStopLossInput = ({
         trade_type: Object.keys(trade_types)[0],
     });
 
+    // We need to exclude tp in case if type === sl and vise versa in limit order to validate them independently
+    if (is_take_profit_input && proposal_req.limit_order?.stop_loss) {
+        delete proposal_req.limit_order.stop_loss;
+    }
+    if (!is_take_profit_input && proposal_req.limit_order?.take_profit) {
+        delete proposal_req.limit_order.take_profit;
+    }
+
     const { data: response } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
-        ['proposal', ...Object.entries(new_values).flat().join('-'), Object.keys(trade_types)[0]],
+        [
+            'proposal',
+            ...Object.entries(new_values).flat().join('-'),
+            Object.keys(trade_types)[0],
+            JSON.stringify(proposal_req),
```

Comment:
Add one more dependency to trigger proposal request, which will contain other trade params values (e.g. stake, which effects tp and sl value)